x-pack/plugins/fleet/server/services/agent_policies/full_agent_policy.ts

```diff
@@ -80,6 +82,23 @@ export async function getFullAgentPolicy(
 
   const sourceUri = await getSourceUriForAgentPolicy(soClient, agentPolicy);
 
+  // Build up an in-memory object for looking up Package Info, so we don't have
+  // call `getPackageInfo` for every single policy, which incurs performance costs
+  const packageInfoCache = new Map<string, PackageInfo>();
```

Comment:
can we concurrently fetch the needed package too? I guess it will also help with performance no?

Reply:
I actually don't know if this would be the right approach, because of how we're using an in-memory cache to save on requests. 

If we fetch all the package info concurrently, we'll enqueue a promise for each package right away because the cache will be empty. This means our requests will likely be sent before the cache contains package info for a given package. Sending out every request "at once" means we can't cache the package info and prevent unnecessary requests.

If we keep this synchronous, we wait until each request comes back before we write to the cache. This means when the next request for a given package comes up in the queue, we can reliably check the cache for its package info.

Reply:
I guess the other approach to refactoring this would be something like 

1. Determine the set of unique packages on the policy
2. Fetch those packages concurrently
3. Pass an object containing all those packages' `PackageInfo` objects around

This might make more sense than the implementation I have here now, so I'll take a look at this.

Reply:
Yes I think the approach of first fetching the package set make sense